### PR TITLE
renamed + update docker-compose, updated k8s manifests

### PR DESCRIPTION
**Pull request by avimoondra**
- renames docker-compose-workflows-with-temporal --> docker-compose-with-temporal
- adds code-executor + workflows-backend (merges in https://github.com/tryretool/retool-onpremise/pull/210)
- updates kubernetes_workflows --> kubernetes-with-temporal
- adds workflow-related services to kubernetes
- corresponds to internal docs change here: https://github.com/tryretool/docs/pull/568

### CODE REVIEW

File: kubernetes-with-temporal.yml/retool-container.yaml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+          name: api
+          ports:
+            - containerPort: 3000
+          resources:
+            limits:
+              memory: 4Gi
+              cpu: 2
+            requests:
+              cpu: 1
+              memory: 1Gi
```
Comment (avimoondra) [inline kubernetes-with-temporal.yml/retool-container.yaml:86]: also updated resource requests/limits, but will revisit soon across https://github.com/tryretool/retool-helm and here.

File: docker-compose.yml [hunk 2]
```diff
@@ -26,19 +27,20 @@ services:
       - backend-network
       - db-connector-network
       - db-ssh-connector-network
-      - workflows-network
```
Comment (avimoondra) [inline docker-compose.yml:29]: not needed since backend-network covers it.

File: docker-compose.yml [hunk 3]
```diff
@@ -116,12 +119,57 @@ services:
...
+    environment:
+      - NODE_OPTIONS=--max_old_space_size=1024
+    networks:
+      - code-executor-network
+    # code-executor uses nsjail to sandbox code execution. nsjail requires
+    # privileged container access.
+    # If your deployment does not support privileged access, you can set this
+    # to false to not use nsjail. Without nsjail, all code is run without
+    # sandboxing within your deployment.
+    privileged: true
```
Comment (avimoondra) [inline docker-compose.yml:167]: open to setting this to false, but leaving this comment here - and will create some more docs around it.

Comment (tanaythan) [reply docker-compose.yml:167]: Let's keep this as true since secure should be the default.

File: kubernetes/retool-container.yaml [hunk 4]
```diff
@@ -63,24 +63,24 @@ spec:
...
           limits:
-            memory: 4096M
+            memory: 4Gi
+            cpu: 2
           requests:
-            cpu: "1"
-            memory: 2048M
-        volumeMounts:
-        - name: retool-pv
-          mountPath: /retool_backend/pv-data
```
Comment (tanaythan) [inline kubernetes/retool-container.yaml:78]: Why is the persistent volume removed here?

**Review by tanaythan — COMMENTED** [review]
Mainly one question about the persistent volume, otherwise LGTM

File: kubernetes/retool-container.yaml (hunk 4 above)
Comment (avimoondra) [reply kubernetes/retool-container.yaml:78]: addressed

**Review by tanaythan — APPROVED** [review]
Mainly one question about the persistent volume, otherwise LGTM

**Pull request merged**